hnix-store-core/src/System/Nix/Hash.hs

```diff
@@ -0,0 +1,30 @@
+{-|
+Description : Trunctions of cryptographic hashes.
+Maintainer  : Shea Levy <shea@shealevy.com>
+-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE CPP #-}
+module System.Nix.Hash where
+
+import Control.Monad (void)
+import Data.Coerce (coerce)
+import qualified Data.ByteString as BS
+import Data.Hashable (Hashable (..))
+import Data.Proxy (Proxy(..))
+import Data.Word (Word8)
+import GHC.TypeLits (Nat, KnownNat, natVal, type (<=))
+import Foreign.Ptr (castPtr, Ptr)
+import Foreign.Marshal.Utils (copyBytes)
+
+data HashAlgorithm = TruncatedSHA256 | MD5
+
+newtype Digest (algo :: HashAlgorithm) = Digest { getDigestBytes :: BS.ByteString }
```

Comment:
I'm confused... How are we meant to construct a Digest? Where is truncation happening?

Reply:
This is just a placeholder until we have an actual API design for Digests. There is no code here for producing a digest because the scope of this PR is limited to simplifying the dependency list and reintroducing a small number of stubs to continue allowing things to compile.

The actual Digest API, which supersedes this one, is in https://github.com/haskell-nix/hnix-store/pull/28, which I think should be reviewed separately since the scope is much larger.

Moving forward with #27 and doing fixup in #28 seems harmless, because all the digest stuff in play prior to #28 is just stubs anyway, but I'm open to alternatives.